backend/store/cache/cache.go

```diff
@@ -0,0 +1,258 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/backend/store"
+	"github.com/sensu/sensu-go/backend/store/etcd"
+	"github.com/sensu/sensu-go/types/dynamic"
+)
+
+// key is used to uniquely identify cache entries
+type key struct {
+	name      string
+	namespace string
+}
+
+// Value contains a cached value, and its synthesized companion.
+type Value struct {
+	Resource corev2.Resource
+	Synth    interface{}
+}
+
+func getCacheKey(resource corev2.Resource) key {
+	return key{
+		namespace: resource.GetObjectMeta().Namespace,
+		name:      resource.GetObjectMeta().Name,
+	}
+}
+
+func getCacheValue(resource corev2.Resource, synthesize bool) Value {
+	v := Value{Resource: resource}
+	if synthesize {
+		v.Synth = dynamic.Synthesize(resource)
+	}
+	return v
+}
+
+// MakeSliceCache ...
+func MakeSliceCache(resources []corev2.Resource, synthesize bool) []Value {
+	cache := make([]Value, len(resources))
```

Comment:
Should you replace `[]Value` with the new `resourceSlice` type?

Reply:
Originally, the `sliceCache` was introduced in order to be able to simply return a slice from the EntityCache. This was because the scheduler needed to access all entities, and it would have been inefficient to constantly create a slice from a map.

I don't know if, in the context of a generic cache, it's worth removing that optimization.